kong/tools/utils.lua

```diff
@@ -661,6 +661,47 @@ _M.hostname_type = function(name)
   return "name"
 end
 
+--- splits an optional ':port' section from a hostname
+-- the port section must be decimal digits only.
+-- brackets ('[]') are peeled off the hostname if present.
+-- if there's more than one colon and no brackets, no split is possible.
+-- on non-parseable input, returns name unchanged,
+-- every string input produces at least one string output.
+-- @param name (string) the string to split.
+-- @return hostname (string)
+-- @return port (as number) or nil if not found
```

Comment:
Better to use the `@tparam` and `@treturn` annotations which allow you to specify the types of the arguments and return values in a normalized way, see other examples in this codebase (e.g. in the PDK documentation)

Reply:
done